html/modules/custom/ghi_menu/ghi_menu.routing.yml

```diff
@@ -0,0 +1,7 @@
+ghi_menu.settings:
+  path: '/admin/config/system/menu-links-with-sublinks-deletion'
+  defaults:
+    _form: '\Drupal\ghi_menu\Form\MenuLinkDeleteWithSublinksSettingsForm'
+    _title: 'Menu links with sublinks deletion settings'
+  requirements:
+    _permission: 'administer site configuration'
```

Comment:
GHI specific settings should ideally go under `/admin/config/ghi` and have a shorter url, maybe something like `/admin/config/ghi/menu`

Reply:
Also, you need to define the menu link to actually show up in the admin toolbar menu using `ghi_menu.links.menu.yml`.